homeassistant/components/device_tracker/__init__.py

```diff
@@ -415,23 +437,27 @@ def hidden(self):
     @asyncio.coroutine
     def async_seen(self, host_name: str=None, location_name: str=None,
                    gps: GPSType=None, gps_accuracy=0, battery: str=None,
-                   attributes: dict=None):
+                   attributes: dict=None, source_type: str=REPORT_SOURCE_GPS):
         """Mark the device as seen."""
+        self.source_type = source_type
         self.last_seen = dt_util.utcnow()
         self.host_name = host_name
         self.location_name = location_name
-        self.gps_accuracy = gps_accuracy or 0
+
         if battery:
             self.battery = battery
         if attributes:
             self._attributes.update(attributes)
 
-        self.gps = None
```

Comment:
Since we are not merging properties any more, lets keep setting this to `None` here as it was